vhdbuilder/packer/configure-windows-vhd.ps1

```diff
@@ -286,14 +286,24 @@ function Get-PrivatePackagesToCacheOnVHD {
         $dir = "c:\akse-cache\private-packages"
         New-Item -ItemType Directory $dir -Force | Out-Null
 
+        $mappingFile = "c:\akse-cache\private-packages\mapping.json"
+        $content = @{}
         $urls = $env:WindowsPrivatePackagesURL.Split(",")
         foreach ($url in $urls) {
             $fileName = [IO.Path]::GetFileName($url.Split("?")[0])
             $dest = [IO.Path]::Combine($dir, $fileName)
 
             Write-Log "Downloading a private package to $dest"
             Download-FileWithAzCopy -URL $URL -Dest $dest
+
+            # Example: v1.29.2-hotfix.2024101-1int.zip
+            $version = $fileName.Split('-')[0].SubString(1)
```

Comment:
nit: do we need to check if the fileName contains `-` before accessing `[0]`? Not sure what's the behavior of accessing a zero length array in powershell.

Reply:
The filename should follow the format of the example. If `$fileName` does not contain `-`, `$fileName.Split('-')[0].SubString(1)` will return the full file name without the first character. We can check the build logs to confirm whether it is set correctly.

Reply:
Have we explored using pester to unit test the powershell in agentbaker?

https://github.com/pester/Pester

This is quite dense in logic, and the consequencers of a bug are quite dramatic on the end user experience

Reply:
@bcho Added tests too.